database/schema/12-swap.sql

```diff
@@ -0,0 +1,18 @@
+CREATE TABLE swap
+(
+    timestamp           TIMESTAMP WITHOUT TIME ZONE NOT NULL REFERENCES block (timestamp),
+    hash                TEXT    NOT NULL REFERENCES transaction (hash),
+    height              BIGINT  NOT NULL REFERENCES block (height),
+
+    /* Msgs */
+    routes              JSONB   NOT NULL DEFAULT '[]'::JSONB,
+    sender              VARCHAR(43) NOT NULL,
+    token_in_denom      TEXT NOT NULL,
```

Comment:
seems `token_in_denom` and `token_out_denom` will be used heavily in queries, maybe create an index for them?